src/NBitcoin/BlockStake.cs

```diff
@@ -202,6 +202,11 @@ public override BlockHeader CreateBlockHeader()
             return new PosBlockHeader();
         }
 
+        public ProvenBlockHeader CreateProvenBlockHeader(PosBlock block)
```

Comment:
I am not sure this is correct, this requires casting to `PosConsensusFactory` for to create a PH, but I guess this is the best we have right now.